pkg/discovery/dtofactory/group_dto_builder.go

```diff
@@ -104,23 +104,29 @@ func (builder *groupDTOBuilder) createGroupsByEntityType(entityGroup *repository
 	}
 
 	// group id for parent group
-	groupId := fmt.Sprintf("%s", entityGroup.GroupId)
+	groupId := entityGroup.GroupId
+	isClusterWideGroup := entityGroup.ParentName == ""
+	if isClusterWideGroup {
+		groupId = fmt.Sprintf("%s::All %ss", entityGroup.GroupId, entityGroup.ParentKind)
+	}
 
 	// resize setting for entities is based on the parent type of the group and if sub groups will be created.
 	// If sub groups are to be created, the resize setting is applied on the sub-groups
 	// Else, the parent level container group has the consistent resize flag set to true
-	createSubGroups := hasSubGroups(entityGroup, entityType)
-
-	resizeFlag := resizeEntityType && consistentResizeGroupType(entityGroup) && !createSubGroups
+	needsSubGroups := needsSubGroups(entityGroup, entityType)
+	isSingleEntityInGroup := len(memberList) <= 1
```

Comment:
Why is this logic needed?

Reply:
> Why is the `isSingleEntityInGroup` logic needed?

@ading1977 we get groups by parent (for example Pod by deployment, and containers by deployment) which have only one member (a Deployment which currently has only 1 replica which has only one container). The group containers by deployment, for this deployment, which has only one member with earlier code will have consistentResize=true. This fixes that. The other approach could be to not create a group at all if it has a single entity in it. We can choose either. I left creating the group as is to be consistent with creating a group for each parent.

Reply:
Just want to confirm, what is the original behaviour when there is only one pod/container in the deployment? If the original behaviour is to set consistentResize=true for deployment with one pod/container, I believe we need to keep that behaviour.

Reply:
It would be less confusing to create groups per parent kind/parent instance even if they contain a single member. 
Meng is right,  since the consistent resize flag is read from the group level, the same rule would apply for group with single members as well for computing the consistentResize flag

Reply:
Done!